src/envoy/transcoding/config.cc

```diff
@@ -82,49 +84,103 @@ Config::Config(const Json::Object& config, Server::Instance& server) {
   if (!descriptor_set.ParseFromIstream(&input)) {
     throw EnvoyException("Unable to parse proto descriptor");
   }
-  log().debug("transcoding filter loaded");
+
   for (const auto& file : descriptor_set.file()) {
     if (descriptor_pool_.BuildFile(file) == nullptr) {
       throw EnvoyException("Unable to parse proto descriptor");
     }
   }
+
+  google::api_manager::PathMatcherBuilder<MethodInfo*> pmb;
+
+  for (const auto& service_name : config.getStringArray("services")) {
+    auto service = descriptor_pool_.FindServiceByName(service_name);
+    for (int i = 0; i < service->method_count(); ++i) {
+      auto method = service->method(i);
+
+      auto method_info = new MethodInfo(method);
+      methods_.emplace_back(method_info);
+      auto http_rule = method->options().GetExtension(google::api::http);
+
+      log().debug("/" + service->full_name() + "/" + method->name());
+      log().debug(http_rule.DebugString());
+
+      switch (http_rule.pattern_case()) {
+        case ::google::api::HttpRule::kGet:
+          pmb.Register("GET", http_rule.get(), http_rule.body(), method_info);
+          break;
+        case ::google::api::HttpRule::kPut:
+          pmb.Register("PUT", http_rule.put(), http_rule.body(), method_info);
+          break;
+        case ::google::api::HttpRule::kPost:
+          pmb.Register("POST", http_rule.post(), http_rule.body(), method_info);
+          break;
+        case ::google::api::HttpRule::kDelete:
+          pmb.Register("DELETE", http_rule.delete_(), http_rule.body(),
+                       method_info);
+          break;
+        case ::google::api::HttpRule::kPatch:
+          pmb.Register("PATCH", http_rule.patch(), http_rule.body(),
+                       method_info);
+          break;
+        case ::google::api::HttpRule::kCustom:
+          pmb.Register(http_rule.custom().kind(), http_rule.custom().path(),
+                       http_rule.body(), method_info);
+          break;
+        default:
+          break;
+      }
+
+      pmb.Register("POST", "/" + service->full_name() + "/" + method->name(),
+                   "", method_info);
+    }
+  }
+
```

Comment:
It is better to log a error if no methods added.  It is to detect config error, such as a typo in the services field or missing services field.